TeachingRecordSystem/tests/TeachingRecordSystem.Core.Dqt.CrmIntegrationTests/QueryTests/CreateTeacherTests.cs

```diff
@@ -0,0 +1,154 @@
+namespace TeachingRecordSystem.Core.Dqt.CrmIntegrationTests.QueryTests;
+
+public class CreateTeacherTests : IAsyncLifetime
+{
+    private readonly CrmClientFixture.TestDataScope _dataScope;
+    private readonly CrmQueryDispatcher _crmQueryDispatcher;
+    private readonly CrmClientFixture _fixture;
+
+    public CreateTeacherTests(CrmClientFixture crmClientFixture)
+    {
+        _fixture = crmClientFixture;
+        _dataScope = crmClientFixture.CreateTestDataScope();
+        _crmQueryDispatcher = _dataScope.CreateQueryDispatcher();
+    }
+
+    public Task InitializeAsync() => Task.CompletedTask;
+
+    public async Task DisposeAsync() => await _dataScope.DisposeAsync();
+
+    [Fact]
+    public async Task QueryExecutesSuccessfully()
+    {
+        // Arrange
+        var firstName = _dataScope.TestData.GenerateFirstName();
+        var middleName = _dataScope.TestData.GenerateMiddleName();
+        var lastName = _dataScope.TestData.GenerateLastName();
+        var email = _dataScope.TestData.GenerateUniqueEmail();
+        var ni = _dataScope.TestData.GenerateNationalInsuranceNumber();
+        var dob = _dataScope.TestData.GenerateDateOfBirth();
+        var trn = await _fixture.GetTrnGenerationApiClient().GenerateTrn();
```

Comment:
There's a `GenerateTrn()` method on `TestData` that could be used here instead of exposing `GetTrnGenerationApiClient()`